From d40eae8f110fb9900e82648a2c44710def9f117d Mon Sep 17 00:00:00 2001
From: Nicholas Clark <nick@ccl4.org>
Date: Fri, 27 Jul 2012 17:36:40 +0200
Subject: [PATCH] Restore microperl, which has been unable to build since January.

microperl was broken by commit 82ad65bb0613be64 on 2012-01-16, which used
IN_LOCALE_COMPILETIME for the first time in the C code. Unlike
IN_LOCALE_RUNTIME, it had no fallback definition for microperl.

Commit f90a9a0230170cc0 on 2012-01-28 added the first use of Strtol(),
which means that microperl now needs to assume that the system has strtol().
(Which is not unreasonable, as it's part of C89).
---
 perl.h       |    1 +
 uconfig.h    |    4 ++--
 uconfig.sh   |    2 +-
 uconfig64.sh |    2 +-
 4 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/perl.h b/perl.h
index a7cd37a..c33c191 100644
--- ./perl.h.orig
+++ ./perl.h
@@ -5319,6 +5319,7 @@ typedef struct am_table_short AMTS;
 #define RESTORE_NUMERIC_STANDARD()	/**/
 #define Atof				my_atof
 #define IN_LOCALE_RUNTIME		0
+#define IN_LOCALE_COMPILETIME		0
 
 #endif /* !USE_LOCALE_NUMERIC */
 
diff --git a/uconfig.h b/uconfig.h
index 967ba2c..f21c9bd 100644
--- ./uconfig.h.orig
+++ ./uconfig.h
@@ -527,7 +527,7 @@
  *	This symbol, if defined, indicates that the strtol routine is available
  *	to provide better numeric string conversion than atoi() and friends.
  */
-/*#define HAS_STRTOL	/ **/
+#define HAS_STRTOL	/**/
 
 /* HAS_STRXFRM:
  *	This symbol, if defined, indicates that the strxfrm() routine is
@@ -4735,5 +4735,5 @@
 
 /* Generated from:
  * d9b6a68b192c72c0695b560fae6e4916b381f55df5fdf7911b0ef434840f092e config_h.SH
- * 2678333a6d6676f8aa39b7011bf5133dce5be453df10a8d04cb3187103a10caf uconfig.sh
+ * 2a7f58d271710106d7cb7ac1d0ea33a1caefa9ad74033041bb9996626786941b uconfig.sh
  * ex: set ro: */
diff --git a/uconfig.sh b/uconfig.sh
index 5bf0575..32f6aa5 100644
--- ./uconfig.sh.orig
+++ ./uconfig.sh
@@ -425,7 +425,7 @@ d_strftime='undef'
 d_strlcat='undef'
 d_strlcpy='undef'
 d_strtod='undef'
-d_strtol='undef'
+d_strtol='define'
 d_strtold='undef'
 d_strtoll='undef'
 d_strtoq='undef'
diff --git a/uconfig64.sh b/uconfig64.sh
index 96e0fd3..3d3ca5a 100644
--- ./uconfig64.sh.orig
+++ ./uconfig64.sh
@@ -426,7 +426,7 @@ d_strftime='undef'
 d_strlcat='undef'
 d_strlcpy='undef'
 d_strtod='undef'
-d_strtol='undef'
+d_strtol='define'
 d_strtold='undef'
 d_strtoll='undef'
 d_strtoq='undef'
-- 
1.7.4.1

